### PR TITLE
Use center property (instead of frame) when repositioning annotation views

### PR DESCRIPTION
**Pull request by julianrex**
This addresses https://github.com/mapbox/mapbox-gl-native/issues/11756.

### CODE REVIEW

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -8,10 +8,14 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 
 * Deprecated `+[NSExpression featurePropertiesVariableExpression]` use `+[NSExpression featureAttributesVariableExpression]` instead. ([#11748](https://github.com/mapbox/mapbox-gl-native/pull/11748))
 
+
+### Annotations
+* Fixed an issue where selecting an onscreen annotation could move the map unintentionally. ([#11731](https://github.com/mapbox/mapbox-gl-native/pull/11731))
+* Fixed an issue where annotation views could become distorted if `rotatesToMatchCamera` is `YES`. ([#11756](https://github.com/mapbox/mapbox-gl-native/issues/11756))
```
Comment (fabian-guerra) [inline platform/ios/CHANGELOG.md]: Shouldn't this blurb refer to `([#11817](https://github.com/mapbox/mapbox-gl-native/pull/11817))` instead?

Comment (julianrex) [reply platform/ios/CHANGELOG.md]: Good question - do we normally refer to the source ticket or the PR in the change log?

Comment (friedbunny) [reply platform/ios/CHANGELOG.md]: We always refer to the PR that fixed the issue.

**Comment by julianrex** [comment]
That's why I added that flag to `ios-sdk-examples` as a simple test for this. But I think we can do better as an integration test, where we can be more precise.

**Pull request merged**